code/ensemble/utils/use-ensemble-criteria.R

```diff
@@ -0,0 +1,60 @@
+# Load models according to inclusion criteria before ensembling
+# 
+# Steps:
+# Currently, models included based on having: 
+# 1. All quantiles
+# 2. 4 horizons 
+# 3. Not manually specified for exclusion 
+# 4. Not the hub ensemble
+# 
+# Returns a list:
+# - "forecasts" = tibble, all forecasts passing inclusion criteria
+# - "criteria" = tibble, model names and criteria assessment
+# 
+library(dplyr)
+library(tibble)
+
+use_ensemble_criteria <- function(forecasts = 
+                                    covidHubUtils::load_forecasts(source = "local_hub_repo",
+                                                                  hub_repo_path = here(),
+                                                                  hub = "ECDC",
+                                                                  forecast_dates = forecast_date),
+                                  exclude_models = NULL,
+                                  team_name = "EuroCOVIDhub",
+                                  forecast_date = floor_date(today(), "week", 1)) {
+  
+  # 1. Identify models with all quantiles
+  quantiles <- round(c(0.01, 0.025, seq(0.05, 0.95, by = 0.05), 0.975, 0.99), 3)
+  
+  all_quantiles <- forecasts %>%
+    group_by(model, target_variable, location, target_end_date) %>%
+    summarise(n_quantiles = n(), 
+              .groups = "drop") %>%
+    filter(n_quantiles != length(quantile)) %>%
```

Comment:
I think there's a nifty bug here. `length(quantile)` will take the `quantile` function and therefore always be length 1. Did you mean to write `n_quantiles == length(quantiles)`?

Also it's probably more robust to summarise `all_quantiles_present = setequal(quantile, quantiles)` (as it checks that the quantiles are actually the same, not just same length).

Reply:
Thanks for spotting! Switched to using `setequal` after dropping point forecasts.